src/test/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTaskTest.java

```diff
@@ -62,8 +58,6 @@
 
   @Test
   public void testPutAndFlush() throws Exception {
-    InternalTestCluster cluster = ESIntegTestCase.internalCluster();
```

Comment:
This ties with the discussion points started in @rhauch comment. This class is more an integration test than a unit test, and as well we where using, more abusing, the embedded client for test. Its is a better practise to spin a "real" Elasticsearch and run the test as is done now with test containers.

In the current shape, we only test with one Elasticsearch version, while introducing the changes proposed in this PR, we will be able to test with multiple versions, current and future. This will provide us with a better test bed, without losing anything.